frame/nfts/src/functions.rs

```diff
@@ -224,6 +240,13 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 		ItemPriceOf::<T, I>::remove(&collection, &item);
 		PendingSwapOf::<T, I>::remove(&collection, &item);
 
+		// NOTE: if item's settings are not empty (e.g. item's metadata is locked)
+		// then we keep the record and don't remove it
+		let settings = Self::get_item_settings(&collection, &item)?;
+		if settings.is_empty() {
+			ItemConfigOf::<T, I>::remove(&collection, &item);
+		}
```

Comment:
Is there an attack here where I mint and burn lots of items getting my deposit back but I rack up settings in the state?